src/state/CallViewModel.ts

```diff
@@ -755,80 +746,76 @@ export class CallViewModel extends ViewModel {
   );
 
   /**
-   * List of MediaItems that we want to display
+   * List of MediaItems that we want to have tiles for.
    */
   private readonly mediaItems$ = this.scope.behavior<MediaItem[]>(
-    combineLatest([this.participantsByRoom$, duplicateTiles.value$]).pipe(
-      scan((prevItems, [participantsByRoom, duplicateTiles]) => {
-        const newItems: Map<string, UserMedia | ScreenShare> = new Map(
-          function* (this: CallViewModel): Iterable<[string, MediaItem]> {
-            for (const {
-              livekitRoom,
-              participants,
-              url,
-            } of participantsByRoom) {
-              for (const { id, participant, member } of participants) {
-                for (let i = 0; i < 1 + duplicateTiles; i++) {
-                  const mediaId = `${id}:${i}`;
-                  const prevMedia = prevItems.get(mediaId);
-                  if (prevMedia instanceof UserMedia)
-                    prevMedia.updateParticipant(participant);
-
-                  yield [
+    generateKeyed$<
+      [typeof this.participantsByRoom$.value, number],
+      MediaItem,
+      MediaItem[]
+    >(
+      combineLatest([this.participantsByRoom$, duplicateTiles.value$]),
+      ([participantsByRoom, duplicateTiles], createOrGet) => {
+        const items: MediaItem[] = [];
+
+        for (const { livekitRoom, participants, url } of participantsByRoom) {
```

Comment:
Do we have some kind of naming guide?
participant: livekit participant
roomMember: matrix room member
member: rtc memebership member

(If we follow the above strictly it might be worth mentioning it explicitly somewhere?)

Reply:
27f24ca5659e309a8ddfb883340effb1b3fdde5b